libs/sdk-core/src/swap_out/reverseswap.rs

```diff
@@ -140,17 +177,24 @@ impl BTCSendSwap {
     /// status persisted.
     pub(crate) async fn create_reverse_swap(
         &self,
-        req: SendOnchainRequest,
+        req: CreateReverseSwapArg,
     ) -> ReverseSwapResult<FullReverseSwapInfo> {
-        Self::validate_rev_swap_args(&req.onchain_recipient_address)?;
+        Self::validate_recipient_address(&req.onchain_recipient_address())?;
 
-        let reverse_routing_node = self
+        let routing_node = self
             .reverse_swapper_api
             .fetch_reverse_routing_node()
-            .await?;
+            .await
+            .map(hex::encode)?;
         let created_rsi = self
-            .create_and_validate_rev_swap_on_remote(req, hex::encode(reverse_routing_node))
+            .create_and_validate_rev_swap_on_remote(req.clone(), routing_node)
             .await?;
+
+        if let CreateReverseSwapArg::V2(req) = req {
+            let claim_fee = created_rsi.onchain_amount_sat - req.receive_amount_sat;
```

Comment:
possible negative value here?

Reply:
This should take care of it: b8cb40df284c05b61b7c3ea67d6e8013ca6ddb5f